src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs

```diff
@@ -453,13 +453,12 @@ public async Task<IActionResult> FileUploadAmendedFile(FileUploadAmendedFileView
         [HttpGet]
         [Route("add/select-journey")]
         [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
-        public IActionResult SelectAddDraftApprenticeshipJourney(SelectAddDraftApprenticeshipJourneyRequest request)
+        public async Task<IActionResult> SelectAddDraftApprenticeshipJourney(SelectAddDraftApprenticeshipJourneyRequest request)
         {
-            var model = new SelectAddDraftApprenticeshipJourneyViewModel
-            {
-                ProviderId = request.ProviderId,
-                IsBulkUploadV2Enabled = _featureTogglesService.GetFeatureToggle(ProviderFeature.BulkUploadV2WithoutPrefix).IsEnabled
-            };
+            var model = await _modelMapper.Map<SelectAddDraftApprenticeshipJourneyViewModel>(request);
+
+            if (!model.HasExistingCohort && !model.HasCreateCohortPermission)
+                return RedirectToAction("Error", "Error", new { statusCode = 405 });
```

Comment:
do we need to redirect to error here? Could simply return the view or redirect to another action. Also 405 doesn't seem like an appropriate status code

Reply:
changed